tests/sources/test_gmail.py

```diff
@@ -119,7 +119,9 @@ def setup_source():
 
 def set_dls_enabled(source, dls_enabled):
     source.set_features(Features({"document_level_security": {"enabled": dls_enabled}}))
-    source.configuration.set_field("use_document_level_security", value=dls_enabled)
+    source.configuration.set_field(
+        "use_document_level_security", field_type="bool", value=dls_enabled
```

Comment:
Intuitively, it would be nice not to pass `field_type` here because the configuration already knows, that `use_document_level_security` is boolean.

Why do we need to pass it here?

Reply:
The `set_field` func has some default values that need to be specified. This isn't an issue in the app because we only run `set_field` when a connector's configuration is first populated. We also co-opt it for tests like this but it's not a simple _make value `x`_.

We could create an alternative func for testing that only updates the value, like `set_value`. WDYT?

Reply:
Actually the more I look into it the more I feel that `set_field` shouldn't be used for managing configurations in testing, so I will create a new func.

Reply:
I've added a new func `set_value` and updated all affected tests.